Homework10/Hw10/Handlers/ExpressionValidator.cs

```diff
@@ -0,0 +1,208 @@
+using Hw10.ErrorMessages;
+using Hw10.Regex;
+
+namespace Hw10.Handlers;
+
+public class ExpressionValidator
+{
+    public static (bool, string) Validate(string expression)
+    {
+        // Check for null or empty
+        if (string.IsNullOrEmpty(expression))
+            return (false, MathErrorMessager.EmptyString);
+
+        // Checking for allowed characters
+        var unknownCharactersCheck = ContainsOnlyAllowedCharacters(expression);
+        if (!unknownCharactersCheck.Item1)
```

Comment:
```suggestion
        if (ContainsOnlyAllowedCharacters(expression) is (false, var error))
```
И убрать то что сверху. Лучше читается, можешь ниже так же сделать